src/ConfigProvider.php

```diff
@@ -34,14 +34,14 @@ public function getDependencyConfig(): array
                 FormAbstractServiceFactory::class,
             ],
             'aliases'            => [
-                Annotation\AnnotationBuilder::class => 'FormAnnotationBuilder',
-                Annotation\AttributeBuilder::class  => 'FormAttributeBuilder',
-                FormElementManager::class           => 'FormElementManager',
+                'FormAnnotationBuilder' => Annotation\AnnotationBuilder::class,
+                'FormAttributeBuilder'  => Annotation\AttributeBuilder::class,
+                'FormElementManager'    => FormElementManager::class,
             ],
             'factories'          => [
-                'FormAnnotationBuilder' => Annotation\BuilderAbstractFactory::class,
-                'FormAttributeBuilder'  => Annotation\BuilderAbstractFactory::class,
-                'FormElementManager'    => FormElementManagerFactory::class,
+                Annotation\AnnotationBuilder::class => Annotation\BuilderAbstractFactory::class,
+                Annotation\AttributeBuilder::class  => Annotation\BuilderAbstractFactory::class,
+                FormElementManager::class           => FormElementManagerFactory::class,
```

Comment:
Note here: swapping aliases and factories has a secondary effect of messing with delegators.

I am not 100% sure if delegators operate the same way on factories or aliases. Consider keeping the priority as it was before :)

Reply:
I asked him to do so: I'd rather check the delegator issue than keeping this forever

Reply:
Reported in https://github.com/laminas/laminas-servicemanager/issues/158

I hope to have a PR there in the next week

Reply:
This is still a BC break then: IMO not worth it for now, or to be extracted and targeted (just this bit) for `4.0.0`

Reply:
It's a BC break only if the current `laminas-servicemanager` major doesn't work with both alias and aliased name

Reply:
If you manage to fix it, we can bump `laminas-servicemanager` to the fix version from laminas/laminas-servicemanager#158 and do it in a minor release.

Removing milestone here meanwhile, since it will be blocked for another while.

Reply:
Why don't I swap the aliases back for now - the rest can be merged preventing future conflicts?

IIRC, delegators simply don't work on aliases and never will: https://github.com/laminas/laminas-servicemanager/pull/91

Reply:
Yeah, I think that would be OK to do meanwhile :+1:

Reply:
@Slamdunk your call here - can also discuss it tonight.